tests/test_utils.py

```diff
@@ -63,6 +63,28 @@ def _assert_keys_not_listed(self, pfx: str, keys: set):
                 time.sleep(6)
         self.assertEqual(0, len(listed_keys.intersection(keys)))
 
+    def test_upload_object(self):
+        with self.subTest("Test upload without verification"):
+            self._test_upload_object(False)
+        with self.subTest("Test upload with verification"):
+            self._test_upload_object(True)
+
+    def _test_upload_object(self, verify: bool):
+        config.object_exists_waiter_config['Delay'] = int(verify)
+        s3_client = boto3.client("s3")
+        tagging = dict(foo="bar", doom="gloom")
+        metadata = dict(sed="awk", perl="ruby")
```

Comment:
`perl != "ruby"`

Ruby is still relevant sir.

Reply:
Ruby, alas, has never been relevant.

Reply:
Unfortunately I think our gitlab CI is written in ruby.